core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -0,0 +1,165 @@
+
+        if (ctx.shouldReportUnhandledExceptions() && !isExpectedException) {
+            if (scheduledUpdater.compareAndSet(this, 0, 1)) {
+                reportingTaskFuture = ctx.eventLoop().scheduleAtFixedRate(
+                        this::reportException, interval.toMillis(), interval.toMillis(), TimeUnit.MILLISECONDS);
+            }
+
+            if (thrownException == null) {
+                thrownException = cause;
+            }
+
+            micrometerCounter.increment();
+            counter.increment();
+        }
+
+        return delegate.onServiceException(ctx, cause);
+    }
+
+    @Nullable
+    @Override
+    public AggregatedHttpResponse renderStatus(ServiceConfig config, @Nullable RequestHeaders headers,
+                                               HttpStatus status, @Nullable String description,
+                                               @Nullable Throwable cause) {
+        return delegate.renderStatus(config, headers, status, description, cause);
+    }
+
+    @Override
+    public void serverStarting(Server server) throws Exception {}
+
+    @Override
+    public void serverStarted(Server server) throws Exception {}
+
+    @Override
+    public void serverStopping(Server server) throws Exception {}
+
+    @Override
+    public void serverStopped(Server server) throws Exception {
+        if (reportingTaskFuture != null) {
+            reportingTaskFuture.cancel(true);
+            reportingTaskFuture = null;
+        }
+    }
+
+    private void reportException() {
+        final long totalExceptionsCount = counter.sum();
+        final long newExceptionsCount = totalExceptionsCount - lastExceptionsCount;
+        if (newExceptionsCount == 0) {
+            return;
+        }
+
+        final Throwable exception = thrownException;
+        if (exception != null) {
+            logger.warn("Observed {} unhandled exceptions in last {}. " +
+                        "Please consider adding the LoggingService decorator to get detailed error logs. " +
+                        "One of the thrown exceptions:",
+                        newExceptionsCount, TextFormatter.elapsed(interval.toNanos()), exception);
+            thrownException = null;
+        } else {
+            logger.warn("Observed {} unhandled exceptions in last {}. " +
+                        "Please consider adding the LoggingService decorator to get detailed error logs.",
+                        newExceptionsCount, TextFormatter.elapsed(interval.toNanos()));
+        }
+
+        lastExceptionsCount = totalExceptionsCount;
+    }
+
+    private static final class MicrometerCounter {
+
+        private final LongAdder counter;
+
+        MicrometerCounter(MeterRegistry registry, MeterIdPrefix idPrefix) {
+            counter = new LongAdder();
+            registry.more().counter(idPrefix.name(), idPrefix.tags(), counter);
+        }
+
+        void increment() {
+            counter.increment();
+        }
+    }
```

Comment:
Isn't this unnecessary now that we keep two counters? We can just do this and let Micrometer decide whether it should count on `.increment()`.

```java
private final Counter micrometerCounter;

// in the constructor:
micrometerCounter = meterRegistry.counter(...);
```

Reply:
Ok, I'll change it to `Counter` type 👍